drivers/hmis/spec/models/hmis/form/processor_spec.rb

```diff
@@ -0,0 +1,125 @@
+  let!(:user) { create(:user).tap { |u| u.add_viewable(ds) } }
+  let(:hmis_user) { Hmis::User.find(user.id)&.tap { |u| u.update(hmis_data_source_id: ds.id) } }
+  let(:hmis_hud_user) { Hmis::Hud::User.from_user(hmis_user) }
+
+  let!(:fd) { create :hmis_form_definition }
+
+  let(:o1) { create :hmis_hud_organization, data_source: ds, user: hmis_hud_user }
+  let(:p1) { create :hmis_hud_project, data_source: ds, organization: o1, user: hmis_hud_user }
+  let(:c1) { create :hmis_hud_client, data_source: ds, user: hmis_hud_user }
+  let!(:e1) { create :hmis_hud_enrollment, data_source: ds, project: p1, client: c1, user: hmis_hud_user }
+
+  it 'ingests EnrollmentCoC into the hud tables' do
+    assessment = Hmis::Hud::Assessment.new_with_defaults(enrollment: e1, user: hmis_hud_user, form_definition: fd, assessment_date: Date.current)
+    assessment.assessment_detail.hud_values = {
+      'EnrollmentCoc.cocCode' => 'MA-507',
+    }
+
+    assessment.assessment_detail.processor.run!
+    assessment.save_not_in_progress
+
+    expect(assessment.enrollment.enrollment_cocs.count).to eq(1)
+    expect(assessment.enrollment.enrollment_cocs.first.coc_code).to eq('MA-507')
+  end
+
+  it 'ingests IncomeBenefit into the hud tables' do
+    assessment = Hmis::Hud::Assessment.new_with_defaults(enrollment: e1, user: hmis_hud_user, form_definition: fd, assessment_date: Date.current)
+    assessment.assessment_detail.hud_values = {
+      'IncomeBenefit.incomeFromAnySource' => 'YES',
+      'IncomeBenefit.earned' => true,
+      'IncomeBenefit.earnedAmount' => 1,
+      'IncomeBenefit.otherInsurance' => false,
+    }
+
+    assessment.assessment_detail.processor.run!
+    assessment.save_not_in_progress
+
+    expect(assessment.enrollment.income_benefits.count).to eq(1)
+
+    income_benefits = assessment.enrollment.income_benefits.first
+    expect(income_benefits.income_from_any_source).to eq(1)
+    expect(income_benefits.earned).to eq(1)
+    expect(income_benefits.earned_amount).to eq(1)
+    expect(income_benefits.other_insurance).to eq(0)
+  end
+
+  it 'ingests DisabilityGroup into multiple Disabilities' do
+    assessment = Hmis::Hud::Assessment.new_with_defaults(enrollment: e1, user: hmis_hud_user, form_definition: fd, assessment_date: Date.current)
+    assessment.assessment_detail.hud_values = {
+      'DisabilityGroup.physicalDisability' => 'YES',
+      'DisabilityGroup.physicalDisabilityIndefiniteAndImpairs' => 'YES',
+      'DisabilityGroup.developmentalDisability' => 'NO',
+      'DisabilityGroup.chronicHealthCondition' => 'YES',
+      'DisabilityGroup.chronicHealthConditionIndefiniteAndImpairs' => 'NO',
+      'DisabilityGroup.hivAids' => 'YES',
+      'DisabilityGroup.mentalHealthDisorder' => 'NO',
+      'DisabilityGroup.substanceUseDisorder' => 'BOTH_ALCOHOL_AND_DRUG_USE_DISORDERS',
+      'DisabilityGroup.substanceUseDisorderIndefiniteAndImpairs' => 'YES',
+      'DisabilityGroup.disablingCondition' => 'YES',
```

Comment:
I think we still need to determine whether 99s should be filled in by the frontend before sending back, or if they should be added by the processor.

For example say someone enters "No" for physical disability, "Yes" for hiv/aids, and leaves everything else blank. It (currently) results in the following
```
  "DisabilityGroup.physicalDisability": "NO",
  "DisabilityGroup.hivAids": "YES",
  "DisabilityGroup.disablingCondition": "YES"
```

* The physical disability record's "IndefiniteAndImpairs" field should be saved as `99`, not `nil`
* Should we still create all 6 disability records (filled with 99s) or should we only created the 2 that were specified?

Definitely for a follow-up discussion/PR, no changes requested on that here.

Reply:
We should definitely create all 6 records with 99s or nils.  I think not having some of them may really impact reporting down the road.